packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -388,7 +388,7 @@ class KotlinGenerator extends StructuredGenerator<KotlinOptions> {
                 final String forceUnwrap =
                     func.returnType.isNullable ? '?' : '';
                 indent.writeln(
-                    'val result = if (it is Int) it.toLong() else it as$forceUnwrap Long');
+                    'val result = if (it is Int) it.toLong() else it as Long$forceUnwrap');
               } else {
                 indent.writeln(
                     'val result = ${_cast('it', kotlinType: returnType, safeCast: func.returnType.isNullable)}');
```

Comment:
Isn't safe casting here (vs force casting to a possibly nullable type) wrong too?

Reply:
Where is safe casting happening?

Reply:
If you mean the name `safeCast` I changed that too

Reply:
> Where is safe casting happening?

When I reviewed it last time, `_cast` was still actually doing safe casting (`as?`) any time `safeCast` was true. Now it looks like it's not which is good, but there's still a parameter called `safeCast` which seems even more confusing. Can't we just remove the parameter entirely now? It's only used in a condition that checks for the case where it would have generated `as? Any`, in order to prevent it, but that's now impossible for it to generate anyway.